feathr_project/test/test_feature_name_validation.py

```diff
@@ -0,0 +1,36 @@
+import string
+
+import pytest
+
+from feathr import FeatureBase, FeatureNameValidationError
+
+
+@pytest.mark.parametrize('bad_feature_name',
+                         [None,
+                          ''])
```

Comment:
also add test for "."

Reply:
This test is actually just testing for empty or None feature names. 

```python
def test_feature_name_fails_on_punctuation_chars():
    for char in set(string.punctuation) - set('_'):
        with pytest.raises(FeatureNameValidationError,  match="only letters, numbers, and underscores are allowed"):
            FeatureBase.validate_feature_name(f"feature_{char}_name")
```

This test checks for "." (which is part of string.punctuation).